HW Docker

alex31425 opened the pull request:
Somehow the SQL result does not save in local folder. I am figuring it out and fixing it.

alex31425 committed:
Add files via upload

michelle-an commented on docker-build/Dockerfile:
```diff
@@ -0,0 +1,21 @@
+
+VOLUME ["/home/mu-ting/docker-build","SQLBASH.sh","baseball.sql"]
+
+CMD ["/home/mu-ting/docker-build","baseball.sql","SQLBASH.sh"]
```
When I try to run docker-compose up I get this error:
ERROR: for sql_service  Cannot start service sql_service: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/home/mu-ting/docker-build\": permission denied": unknown

I don't think this has anything to do with a directory issue. It should be creating that directory in the container and it makes it past the WORKDIR step on line 9. 

If you have any hints that I might be doing wrong please tell me so I can actually get this to run. I think it's more of an issue on my end, because just by looking at it I can't figure out what wouldn't be working.

alex31425 replied:
I think you should update the PATH where you save this file. I saved the files in this path so try update the PATH

michelle-an commented on docker-build/SQLBASH.sh:
```diff
@@ -0,0 +1,7 @@
+
+
+
+mysql -u root -proot < /home/mu-ting/docker-build/baseball.sql > output.txt
```
This method is actually a lot cleaner than what I tried. I ended up making a lot of little mysql connections to run each part of my script,  but a few times it would fail to connect at a step and I would have only part of the calculation done. I think making one connection and running everything at once was a better way to go.

michelle-an commented on docker-build/baseball.sql:
```diff
@@ -0,0 +1,42 @@
+use baseball;
+#select * from game limit 10;
+
+drop table if exists batting_avg_wDATEDIFF ;
```
Don't drop the tables! instead you should use "create table if not exists" to make the intermediate tables. That way if the tables still exist from the last calculation it will run faster.

michelle-an commented on docker-build/baseball.sql:
```diff
@@ -0,0 +1,42 @@
+
+insert into batting_avg_wDATEDIFF (
+game_id,batter,hit,atbat,local_date,diff_day)
+select g.game_id,bc.batter,bc.hit,bc.atBat,g.local_date,DATEDIFF('2012-07-05 00:00:00' ,g.local_date) as date_diff
```
When you call this table later, you never actually use this date_dif column. I think you can get rid of it and save yourself a bunch of lines.

michelle-an commented on docker-build/baseball.sql:
```diff
@@ -0,0 +1,42 @@
+    
+# select * from batting_avg_wDATEDIFF bawd;
+
+set @Enddate = '2008-08-31';
```
I don't think you ever actually use this variable?

michelle-an commented on docker-build/baseball.sql:
```diff
@@ -0,0 +1,42 @@
+
+set @Enddate = '2008-08-31';
+
+drop table if exists rolling;
```
Same thing as above about create table if not exists.

michelle-an commented on docker-build/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+    restart: unless-stopped
+    depends_on: 
+     - db
+    stdin_open: true
```
I tried removing this line to see if it would help me get this to run, but nothing changed. The line is probably fine, I just didn't recognize it from mine so I thought it would be a good starting change to try.

michelle-an commented on docker-build/docker-compose.yml:
```diff
@@ -0,0 +1,38 @@
+  db:
+    image: mariadb:10.4
+    restart : always
+    ports:
```
I ended up removing this port assignment because it was having some problems on my computer. It might have just been an issue with 3306 being used for some other service on my end.

michelle-an reviewed (approved):
I tried my best to get this to run but I kept hitting the same error. I'll keep trying later and see what I can do. But I wanted to post comments now so I could at least give some feedback.

I don't think it's a problem with how something on my end is set up. Once I get it working I'll update with what change I made.

alex31425 committed:
Updated SQL file

dafrenchyman committed:
Fixed everything

Please check canvas for detailed notes, but I had a lot of stuff I had to correct

dafrenchyman committed:
Added some stuff to ignore
